custom_model_runner/datarobot_drum/drum/model_adapter.py

```diff
@@ -103,12 +103,21 @@ def load_model_from_artifact(self):
     def _load_model_via_hook(self):
         self._logger.debug("Load model hook will be used to load the model")
         # noinspection PyCallingNonCallable
-        model = self._custom_hooks[CustomHooks.LOAD_MODEL](self._model_dir)
-        if model:
-            self._logger.debug("Model was successfully loaded by load hook")
-            return model
+
+        try:
+            model = self._custom_hooks[CustomHooks.LOAD_MODEL](self._model_dir)
+        except Exception as exc:
+            raise type(exc)(
```

Comment:
Is this good practice?

Reply:
Looks sketchy to me.

Reply:
```# for Python 3.x
...
def bar(arg1):
    try:
        foo()
    except Exception as e:
        import sys
        raise type(e)(str(e) +
                      ' happens at %s' % arg1).with_traceback(sys.exc_info()[2])```

Reply:
is this better?

Reply:
I'm not sure about re-raising exception of the same type though

Reply:
https://stackoverflow.com/questions/6062576/adding-information-to-an-exception

Reply:
yes this is better - I don't need a new exception with the exception chaining I did
testing how tracebacks will look for your suggestion

Reply:
pls review again: https://github.com/datarobot/datarobot-user-models/pull/101/commits/1251dab5c76d79d19e377bd0b034c9b1411be133

now we have one exception (instead of 2) and one traceback:
```
2020-08-03 20:53:29,982 ERROR drum.datarobot_drum.drum.runtime:  Traceback (most recent call last):
  File "custom_model_runner/datarobot_drum/drum/main.py", line 93, in main
    CMRunner(runtime).run()
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/drum.py", line 225, in run
    self._run_fit_and_predictions_pipelines_in_mlpiper()
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/drum.py", line 411, in _run_fit_and_predictions_pipelines_in_mlpiper
    _pipeline_executor.init_pipeline()
  File "/Users/roman/.virtualenvs/drum2/lib/python3.7/site-packages/mlpiper/pipeline/executor.py", line 252, in init_pipeline
    self._init_pipeline()
  File "/Users/roman/.virtualenvs/drum2/lib/python3.7/site-packages/mlpiper/pipeline/executor.py", line 223, in _init_pipeline
    system_conf, ee_conf, self._ml_engine
  File "/Users/roman/.virtualenvs/drum2/lib/python3.7/site-packages/mlpiper/pipeline/dag.py", line 117, in configure_connected_pipeline
    dag_node.component_runner.configure(input_args)
  File "/Users/roman/.virtualenvs/drum2/lib/python3.7/site-packages/mlpiper/pipeline/component_runner/python_connected_component_runner.py", line 10, in configure
    self._dag_node.main_cls().configure(params)
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/resource/components/Python/prediction_server/prediction_server.py", line 68, in configure
    self._predictor.configure(params)
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py", line 27, in configure
    self._model = self._model_adapter.load_model_from_artifact()
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/model_adapter.py", line 92, in load_model_from_artifact
    self._model = self._load_model_via_hook()
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/model_adapter.py", line 112, in _load_model_via_hook
    ).with_traceback(sys.exc_info()[2]) from None
  File "/Users/roman/workspace/datarobot-user-models/custom_model_runner/datarobot_drum/drum/model_adapter.py", line 108, in _load_model_via_hook
    model = self._custom_hooks[CustomHooks.LOAD_MODEL](self._model_dir)
  File "/Users/roman/workspace/cm_testing/load_model_raises/model/custom.py", line 4, in load_model
    raise ValueError('exception')
ValueError: Model loading hook failed to load model: exception

```

Reply:
(`from None` is required so that we don't get "During handling of the above exception, another exception occurred" https://stackoverflow.com/questions/52725278/during-handling-of-the-above-exception-another-exception-occurred)